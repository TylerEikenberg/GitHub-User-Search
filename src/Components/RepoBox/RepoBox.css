.RepoBox-container {
  width: 90vw;
  background-color: var(--sub-color);
  border-radius: 5px;
  display: flex;
  flex-direction: column;
  margin-top: 3vh;
  padding: 5px 0 5px 0;
  -webkit-box-shadow: 0px 5px 5px 2px rgba(168, 194, 202, 1);
  -moz-box-shadow: 0px 5px 5px 2px rgba(168, 194, 202, 1);
  box-shadow: 0px 5px 5px 2px rgba(168, 194, 202, 1);
  text-decoration: none;
  transition: background-color 0.5s;
}
.RepoBox-container:hover {
  background-color: var(--sub-color-hover);
}
.RepoBox-repo-link {
  color: var(--link-color);
  margin-left: 3vw;
  text-decoration: none;
}

.RepoBox-stats-container {
  display: flex;
  justify-content: space-around;
}

@media only screen and (min-width: 1024px) {
  .RepoBox-container {
    max-width: 50vw;
    background-color: #edf2f8;
    border-radius: 5px;
    display: flex;
    flex-direction: column;
    margin-top: 3vh;
    padding: 5px 0 5px 0;
    -webkit-box-shadow: 0px 5px 5px 2px rgba(168, 194, 202, 1);
    -moz-box-shadow: 0px 5px 5px 2px rgba(168, 194, 202, 1);
    box-shadow: 0px 5px 5px 2px rgba(168, 194, 202, 1);
  }
}
