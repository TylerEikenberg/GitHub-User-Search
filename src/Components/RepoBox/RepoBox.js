import React from "react";
import { StatBox } from "../";
import "./RepoBox.css";

function RepoBox({
  repoData = {
    html_url: "",
    name: "",
    stargazers_count: 0,
    forks: 0,
    watchers: 0
  }
}) {
  return (
    <div data-testid="RepoBox-div" className="RepoBox-container">
      <span>
        <a
          data-testid="RepoBox-a"
          className="RepoBox-repo-link"
          href={repoData.html_url}
        >
          {repoData.name}
        </a>
      </span>
      <span data-testid="RepoBox-span" className="RepoBox-stats-container">
        <StatBox number={repoData.stargazers_count} stat={"Stars"} />
        <StatBox number={repoData.forks} stat={"Forks"} />
        <StatBox number={repoData.watchers} stat={"Watchers"} />
      </span>
    </div>
  );
}

export default RepoBox;
