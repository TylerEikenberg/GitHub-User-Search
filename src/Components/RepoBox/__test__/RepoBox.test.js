import RepoBox from "../RepoBox";
import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import "@testing-library/jest-dom/";

afterEach(cleanup);

it("RepoBox renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<RepoBox></RepoBox>, div);
});
it("RepoBox has className RepoBox-container", () => {
  const { getByTestId } = render(<RepoBox></RepoBox>);
  expect(getByTestId("RepoBox-div")).toHaveClass("RepoBox-container");
});
it("RepoBox has className RepoBox-repo-link", () => {
  const { getByTestId } = render(<RepoBox></RepoBox>);
  expect(getByTestId("RepoBox-a")).toHaveClass("RepoBox-repo-link");
});
it("RepoBox has className RepoBox-stats-container", () => {
  const { getByTestId } = render(<RepoBox></RepoBox>);
  expect(getByTestId("RepoBox-span")).toHaveClass("RepoBox-stats-container");
});
it("RepoBox, matches snapshot", () => {
  const tree = TestRenderer.create(<RepoBox></RepoBox>).toJSON();
  expect(tree).toMatchSnapshot();
});
