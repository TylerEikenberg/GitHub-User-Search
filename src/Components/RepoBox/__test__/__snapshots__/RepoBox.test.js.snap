// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoBox, matches snapshot 1`] = `
<div
  className="RepoBox-container"
  data-testid="RepoBox-div"
>
  <span>
    <a
      className="RepoBox-repo-link"
      data-testid="RepoBox-a"
      href=""
    >
      
    </a>
  </span>
  <span
    className="RepoBox-stats-container"
    data-testid="RepoBox-span"
  >
    <div
      className="StatBox-container"
    >
      <h3
        className="StatBox-h3 StatBox-top"
      >
        0
      </h3>
      <h3
        className="StatBox-h3 StatBox-bottom"
      >
        Stars
      </h3>
    </div>
    <div
      className="StatBox-container"
    >
      <h3
        className="StatBox-h3 StatBox-top"
      >
        0
      </h3>
      <h3
        className="StatBox-h3 StatBox-bottom"
      >
        Forks
      </h3>
    </div>
    <div
      className="StatBox-container"
    >
      <h3
        className="StatBox-h3 StatBox-top"
      >
        0
      </h3>
      <h3
        className="StatBox-h3 StatBox-bottom"
      >
        Watchers
      </h3>
    </div>
  </span>
</div>
`;
